### PR TITLE
Add deprecated comments to rn

### PR DESCRIPTION
**Pull request by ostolero**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-822)

## Description
Added specific release notes comments for deprecated Integration, scripts, and playbook.
The comment will be added automatically when the `deprecate: true` flag is enabled in yml.

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 1]
```diff
@@ -27,8 +27,42 @@
...
+
+def commands_to_deprecate_dict(commands: dict):
+    res_dict = {}
+    for command in commands:
+        res_dict[command.get('name')] = command.get('deprecated')
+
+    return res_dict
+
+
+def get_deprecated_custom_rn(path: str, file_type):
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:46]: what does custom mean in this name?

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 2]
```diff
@@ -27,8 +27,42 @@
...
 
+from demisto_sdk.commands.common.content.objects.pack_objects import \
+    Integration, Script, Playbook
+
 json = JSON_Handler()
 
+CLASS_BY_FILE_TYPE = {FileType.INTEGRATION: Integration, FileType.SCRIPT: Script, FileType.PLAYBOOK: Playbook}
+
+
+def commands_to_deprecate_dict(commands: dict):
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:38]: lets return a set here of only the deprecated items

**Review by JudahSchwartz — COMMENTED** [review]
Very nice!
Will wait for unit tests for approval

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 1 above)
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:46]: Haha not sure, will remove it.

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 3]
```diff
@@ -29,6 +31,56 @@
...
+
+
+def commands_to_deprecate_set(commands: list) -> set:
+    """return a set of the deprecated commands only"""
+    res_set = set()
+    for command in commands:
+        if command.get('deprecated'):
+            res_set.add(command.get('name'))
+
+    return res_set
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:44]:
```suggestion
    return { command.get('name') for command in commands if command.get('deprecated') }
```
Style choice =)

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 4]
```diff
@@ -29,6 +31,56 @@
...
+    """
+    old_yml_obj = get_remote_file(path)
+    new_yml_obj = CLASS_BY_FILE_TYPE[file_type](path)
+
+    return old_yml_obj, new_yml_obj
+
+
+def get_deprecated_rn(path: str, file_type):
+    """Generate rn for deprecated items"""
+    old_yml, new_yml = get_yml_objects(path, file_type)
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:65]: do we care that these are yml objects? content_items seems a bit more clear to me

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 5]
```diff
@@ -548,7 +600,13 @@ def build_rn_desc(self, _type: FileType = None, content_name: str = '', desc: st
                 if self.update_type == 'documentation':
                     rn_desc += '- Documentation and metadata improvements.\n'
                 else:
-                    rn_desc += f'- {text or "%%UPDATE_RN%%"}\n'
+                    deprecate_rn = ''
+                    if _type in (FileType.INTEGRATION, FileType.SCRIPT, FileType.PLAYBOOK):
+                        deprecate_rn = get_deprecated_rn(path, _type)
+                    if deprecate_rn:
+                        rn_desc += f'- {text}\n{deprecate_rn}'
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:607]: if text isnt provided we'll have an extra blank line here. Lets add
if text:
    rn_desc += f'{text}\n'
rn_desc +=deprecate_rn

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 6]
```diff
@@ -29,6 +31,56 @@
...
+    new_yml_obj = CLASS_BY_FILE_TYPE[file_type](path)
+
+    return old_yml_obj, new_yml_obj
+
+
+def get_deprecated_rn(path: str, file_type):
+    """Generate rn for deprecated items"""
+    old_yml, new_yml = get_yml_objects(path, file_type)
+
+    if not old_yml.get('deprecated') and new_yml.is_deprecated:
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:67]:
```suggestion
    if not old_yml.is_deprecated and new_yml.is_deprecated:
```

**Review by JudahSchwartz — COMMENTED** [review]
very nice! Great tests

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 7]
```diff
@@ -29,6 +31,56 @@
...
+    """return a set of the deprecated commands only"""
+    res_set = set()
+    for command in commands:
+        if command.get('deprecated'):
+            res_set.add(command.get('name'))
+
+    return res_set
+
+
+def get_yml_objects(path: str, file_type) -> Tuple[Any, Union[Integration, Script, Playbook]]:
```
[the remark this answers is not in the document]
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:47]: old_yml is CommentedMap, Changed to YAMLContentObject

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 6 above)
Comment (ostolero) [reply demisto_sdk/commands/update_release_notes/update_rn.py:67]: Not possible as old_yml is CommentedMap

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 8]
```diff
@@ -34,17 +36,28 @@
...
+        description: The yml description
 
-    return res_set
+    Returns:
+        If a deprecated description is found return it for rn.
+    """
+    deprecate_line_with_replacement = re.findall(DEPRECATED_REGEXES[0], description)
+    deprecate_line_no_replacement = re.findall(DEPRECATED_REGEXES[1], description)
+
+    deprecate_line = deprecate_line_with_replacement + deprecate_line_no_replacement
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:51]: arent deprecate_line_with_replacement and deprecate_line_no_replacement lists?

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 9]
```diff
@@ -34,17 +36,28 @@
 CLASS_BY_FILE_TYPE = {FileType.INTEGRATION: Integration, FileType.SCRIPT: Script, FileType.PLAYBOOK: Playbook}
 
 
-def commands_to_deprecate_set(commands: list) -> set:
-    """return a set of the deprecated commands only"""
-    res_set = set()
-    for command in commands:
-        if command.get('deprecated'):
-            res_set.add(command.get('name'))
+def get_deprecated_comment_from_desc(description: str) -> str:
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:39]: Please add a test

File: demisto_sdk/commands/update_release_notes/update_rn.py [hunk 10]
```diff
@@ -29,6 +33,69 @@
...
+        description: The yml description
+
+    Returns:
+        If a deprecated description is found return it for rn.
+    """
+    deprecate_line_with_replacement = re.findall(DEPRECATED_REGEXES[0], description)
+    deprecate_line_no_replacement = re.findall(DEPRECATED_REGEXES[1], description)
+
+    deprecate_line = deprecate_line_with_replacement + deprecate_line_no_replacement
+    return deprecate_line[0] if deprecate_line else ''
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/update_release_notes/update_rn.py:52]:
```suggestion
    return re.findall(DEPRECATED_REGEXES[0], description) or re.findall(DEPRECATED_REGEXES[1], description) or ''
```

Comment (JudahSchwartz) [reply demisto_sdk/commands/update_release_notes/update_rn.py:52]: I think this is a bit more clear, not a must obviously

**Review by JudahSchwartz — APPROVED** [review]
Great work! Please remember to add the test

File: demisto_sdk/commands/update_release_notes/update_rn.py (hunk 10 above)
Comment (JudahSchwartz) [reply demisto_sdk/commands/update_release_notes/update_rn.py:52]: Nevermind

**Pull request merged**